src/NuGetGallery.Core/Services/SimpleCloudBlobExtensions.cs

```diff
@@ -0,0 +1,78 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Net;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.WindowsAzure.Storage;
+
+namespace NuGetGallery
+{
+    public static class SimpleCloudBlobExtensions
+    {
+        /// <summary>
+        /// Retrieves the blob contents as a string assuming UTF8 encoding if the blob exists.
+        /// </summary>
+        /// <param name="blob">Blob reference.</param>
+        /// <returns>The text content of the blob or null if the blob does not exist.</returns>
+        public static async Task<string> DownloadTextIfExistAsync(this ISimpleCloudBlob blob)
```

Comment:
What do you think of this name? I don't feel strongly either ways


```suggestion
        public static async Task<string> DownloadTextIfExistsAsync(this ISimpleCloudBlob blob)
```

Reply:
Same comment for `OpenReadIfExistAsync`